tests/python/relay/test_ir_nodes.py

```diff
@@ -32,44 +44,58 @@ def test_tensor_type():
     assert tt.span == None
     str(tt)
 
+    # roundtrip preserves alpha-equality
+    back = json_roundtrip(tt)
+    assert back == tt
+
 
 def test_type_param():
     tp = relay.TypeParam('name', relay.Kind.Type)
     assert tp.kind == relay.Kind.Type
     # assert tp.span  # TODO allow us to set span
     str(tp)
 
+    back = json_roundtrip(tp)
+    # pointer equality will not be preserved so alpha-equality will fail
+    assert back.kind == tp.kind
+
 
 def test_func_type():
     type_params = tvm.convert([])
     type_constraints = tvm.convert([])  # TODO: fill me in
     arg_types = tvm.convert([])
-    ret_type = None
+    ret_type = relay.TensorType((1, 2, 3), 'float32')
```

Comment:
Should alpha-equality for function types and function nodes be able to handle having "None" for the ret type? Currently it fails and there is no comment in the header files that says that the field should be nullable. It would not be hard to change alpha-equality to accommodate nulls, as long as we are clear about what is expected.